src/client.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  Chat,
+  ParsedResult,
+  StreamQueryConfig,
+  StreamUpdate,
+  StreamUpdateHandler,
+} from "./types";
+import { deserializeSearchResponse } from "./deserializeSearchResponse";
+import { SNIPPET_START_TAG, SNIPPET_END_TAG } from "./constants";
+
+const DEFAULT_ENDPOINT = "api.vectara.io";
+
+export const streamQuery = async (
+  config: StreamQueryConfig,
+  onStreamUpdate: StreamUpdateHandler
+) => {
+  const requestHeaders = {
+    "x-api-key": config.apiKey,
+    "customer-id": config.customerId,
+    "Content-Type": "application/json",
+  };
+
+  const lambda =
+    typeof config.queryValue === "undefined" ||
+    config.queryValue.trim().split(" ").length > config.hybridNumWords
+      ? config.hybridLambdaLong
+      : config.hybridLambdaShort;
+  const corpusKeyList = config.corpusIds.map((id) => {
+    return {
+      customerId: config.customerId,
+      corpusId: id,
+      lexicalInterpolationConfig: {
+        lambda: lambda,
+      },
+      metadataFilter: config.filter
+        ? `doc.source = '${config.filter}'`
+        : undefined,
+    };
+  });
+
+  const requestBody = JSON.stringify({
+    query: [
+      {
+        query: config.queryValue,
+        start: 0,
+        numResults: config.rerank ? config.rerankNumResults : 10,
+        corpusKey: corpusKeyList,
+        contextConfig: {
+          sentencesBefore: config.summaryNumSentences ?? 2,
+          sentencesAfter: config.summaryNumSentences ?? 2,
+          startTag: SNIPPET_START_TAG,
+          endTag: SNIPPET_END_TAG,
+        },
+        summary: [
+          {
+            responseLang: config.language,
+            maxSummarizedResults: config.summaryNumResults,
+            summarizerPromptName: config.summaryPromptName,
+            chat: {
+              store: config.chat?.store ?? false,
+              conversationId: config.chat?.conversationId,
+            },
+          },
+        ],
+        ...(config.rerank
+          ? {
+              rerankingConfig: {
+                rerankerId: config.rerankerId,
+                ...(config.rerankerId === 272725718
```

Comment:
Can we extract this reranker ID value into a variable to provide some more semantics to the reader?